### PR TITLE
Things-Gateway changes

### PR DESCRIPTION
**Pull request by martinichka**
Changes the are needed for the Things-Gateway. Mainly heap issues. Embedded heap doesn't like null pointers in free(). And stored a string in the session to be able to free it after un-subscribe or disconnect.

### CODE REVIEW

File: src/network.h [hunk 1]
```diff
@@ -11,13 +11,13 @@
...
-#define KEEP_ALIVE_INTERVAL 20
+#define KEEP_ALIVE_INTERVAL 60
 #define COMMAND_TIMEOUT 2000
 #define READ_BUFFER_SIZE 512
 #define SEND_BUFFER_SIZE 512
 
-#define QOS_STATUS QOS1
-#define QOS_DOWN QOS1
+#define QOS_STATUS QOS0
+#define QOS_DOWN QOS0
```
Comment (johanstokking) [inline src/network.h]: Leave these out of this PR

File: src/connector.c [hunk 2]
```diff
@@ -72,7 +75,8 @@ int ttngwc_connect(TTN *s, const char *host_name, int port, const char *key) {
 #if SEND_DISCONNECT_WILL
   Types__DisconnectMessage will = TYPES__DISCONNECT_MESSAGE__INIT;
   will.id = session->id;
-  will.key = (char *)key;
+  if (session->key)
+    will.key = session->key;
```
Comment (johanstokking) [inline src/connector.c:77]: Add braces

**Review by johanstokking — CHANGES_REQUESTED** [review]
Thanks! Few things. Also please remove the comments, the code speaks for itself 😊

File: src/connector.c (hunk 2 above)
Comment (martinichka) [reply src/connector.c:77]: I did it in 'style' of the existing code ;) But I will add. To do it for all other ifs without braces as well?

File: src/network.h (hunk 1 above)
Comment (martinichka) [reply src/network.h]: Keep alive interval as well?

File: src/connector.c (hunk 2 above)
Comment (johanstokking) [reply src/connector.c:77]: Ah my bad! No braces indeed

**Review by johanstokking — CHANGES_REQUESTED** [review]

**Review by johanstokking — APPROVED** [review]

File: src/connector.c (hunk 2 above)
Comment (martinichka) [reply src/connector.c:77]: Reverted the braces :)

Comment (martinichka) [reply src/connector.c:77]: All ok now?

**Pull request merged**